hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/replication/TestVerifyReplication.java

```diff
@@ -447,6 +447,36 @@ public void testVerifyRepJobWithPeerTableNameAndSnapshotSupport() throws Excepti
     checkRestoreTmpDir(CONF2, tmpPath2, 2);
   }
 
+  @Test
+  public void testVerifyReplicationThreadedRecompares() throws Exception {
```

Comment:
Can you tell me how this test verifies the new behavior?

Reply:
The idea here is that the default behavior represents VR the way it currently exists. Constant sleep time, 1 recompare, singly threaded. By creating a VR run that's threaded, and with reCompareTries > 1, we test out the asynchronous comparisons and make sure that the job produces the counters we're expecting.

That being said, we don't actually check that the recompares have been executed `Counters.RE_COMPARES`, which might be a good thing to check.

Reply:
ok so it looks like we're setting up this test to purposefully show validation errors. in which case, we should definitely expect recompares to happen and those recompares should all fail. So yea, I think we should add some assertions around the various recompare-related counters.

Reply:
Sounds good, added those!